### DISCUSSION TITLE
How can I make Microsoft.SemanticKernel.Memory.SearchAsync faster?

### DISCUSSION THREAD

**Discussion by superfamousdog** [microsoft/semantic-kernel#7420] [Q&A] [answered]
I have a .sqlite db with 2,500 rows that I'm using as a memory store:
```
var memoryStore = await SqliteMemoryStore.ConnectAsync(<path to db>);

var memoryBuilder = new MemoryBuilder();
var openAiEmbedder = new OpenAITextEmbeddingGenerationService("text-embedding-3-small",
				_openAiKey);
memoryBuilder.WithTextEmbeddingGeneration(openAiEmbedder);
memoryBuilder.WithMemoryStore(memoryStore);
myItem.Memory = memoryBuilder.Build();
```
when I later perform `myItem.Memory.SearchAsync(collectionName,
				query,
				limit: 1000,
				minRelevanceScore: 0.73);`, it takes a couple seconds to run.
				
Is there a way to speed this up, such as using a different memory store? Have I overlooked anything else?

**Comment by dmytrostruk** [comment, marked as answer]
@gusmally I would recommend using other memory store, since current implementation for Sqlite fetches all DB records during search operation which is not the best approach in terms of performance:
https://github.com/microsoft/semantic-kernel/blob/305b2ee28b6d119b9e2fa63bd1201a90e2b9a29b/dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteMemoryStore.cs#L136-L147

Please take a look at other available memory connectors:
https://github.com/microsoft/semantic-kernel/tree/main/dotnet/src/Connectors

We are also working on new version of memory connectors that will allow to use custom schema and are up to date. This work is happening in feature branch at the moment and experimental version of it will be available soon:
https://github.com/microsoft/semantic-kernel/tree/feature-memory-service

**Reply by superfamousdog** [reply]
Thank you for the analysis and recommendation!
